Exercise/src/my_string.cpp

```diff
@@ -18,49 +22,143 @@ char* Exercise_1::CMyString::TryToAllocate(const size_t nLength) noexcept {
 }
 
 std::size_t Exercise_1::CMyString::GetStringLength(const char *cszStringToCount) noexcept {
+	if(cszStringToCount == nullptr)
+	{
+		return 0;
+	}
+
 	std::size_t nStringLength = 0;
-	char cCurrentChar = cszStringToCount[0];
-	while(cCurrentChar != '\0')
+	char chCurrentChar = cszStringToCount[0];
+	while(chCurrentChar != '\0')
 	{
 		++nStringLength;
-		cCurrentChar = cszStringToCount[nStringLength];
+		chCurrentChar = cszStringToCount[nStringLength];
 	}
 	// Adding 1 because string is zero terminated
 	return nStringLength + 1;
 }
 
-void Exercise_1::CMyString::CopyString(const char *cszStringToCopy, char *szStringToPaste, std::size_t nCountOfChars)
+void Exercise_1::CMyString::CopyString(const char *cszStringToCopy)
 {
-	for(std::size_t nIndex = 0; nIndex < nCountOfChars; nIndex++)
+	if(cszStringToCopy == nullptr)
 	{
-		szStringToPaste[nIndex] = cszStringToCopy[nIndex];
+		return;
 	}
+
+	std::size_t nIndex = 0;
+	while(cszStringToCopy[nIndex] != '\0')
+	{
+		m_szData[nIndex] = cszStringToCopy[nIndex];
+		++nIndex;
+	}
+	m_szData[nIndex] = '\0';
 }
 
-Exercise_1::CMyString::CMyString(const char* cszCharsSequence) 
+void Exercise_1::CMyString::AppendToString(const char *cszStringToAppend)
 {
-	m_nSize = GetStringLength(cszCharsSequence);
-	m_szData = TryToAllocate(m_nSize);
+	if(cszStringToAppend == nullptr)
+	{
+		return;
+	}
+	
+	ReinitializeAndCopy(m_szData, size() + GetStringLength(cszStringToAppend) - 1);
+	
 
-	CopyString(cszCharsSequence, m_szData, m_nSize);
+	size_t nTerminantIndex = GetStringLength(m_szData) - 1;
```

Comment:
Is it old terminationIndex before reallocating?

Reply:
Fixed this stupid error, now it uses m_nSize